src/anomalib/training/learning_strategies/default/predict.py

```diff
@@ -0,0 +1,94 @@
+"""Predict loop."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any
+
+import torch
+from lightning_fabric.utilities import move_data_to_device
+from pytorch_lightning.loops.dataloader.prediction_loop import PredictionLoop
+from pytorch_lightning.loops.epoch.prediction_epoch_loop import PredictionEpochLoop
+
+import anomalib.training as core
+
+
+class AnomalibPredictionEpochLoop(PredictionEpochLoop):
+    """Predict epoch loop."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.trainer: core.AnomalibTrainer
+
+    def _predict_step(self, batch: Any, batch_idx: int, dataloader_idx: int) -> None:
+        """Overrides the predict step of the base class.
+
+        Everything is same except for the custom methods called after trainer's predict step.
+
+        Note:
+            Update methods only between the commented block.
+
+        Warning:
+            The order of callbacks and lightning module's ``predict_step`` has been swapped. Originally callbacks were
+            called before the lightning step. But, we need to compute masks and labels before visualizer callback is
+            called. So, in this case the lightning module's ``predict_step`` is called first, then custom methods for
+            post-processing, and finally, the callbacks are called.
```

Comment:
The problem is that the `lightning_module_hook` gets the anomaly masks that are used by the `post_processor`. So, we cannot call it before the `self.trainer._call_lightning_module_hook`. But since visualizer is the bottleneck here we can consider moving the visualizer callback into custom loops.

Reply:
But the only constraint here is that the post processor and normalizer should be called before the callback hooks right? So the order in my suggestion should address the visualization issue and also keep the ordering of lightning module vs callbacks consistent.